examples/examples/z_bytes.rs

```diff
@@ -11,163 +13,110 @@
-    let input: [i32; 4] = [1, 2, 3, 4];
-    let payload = ZBytes::from_iter(input.iter());
-    for (idx, value) in payload.iter::<i32>().enumerate() {
-        assert_eq!(input[idx], value.unwrap());
-    }
-
-    // Iterator RAW
-    let input: [i32; 4] = [1, 2, 3, 4];
-    let payload = ZBytes::from_iter(input.iter());
-    for slice in payload.slices() {
-        println!("{:02x?}", slice);
-    }
-
-    // HashMap
-    let mut input: HashMap<usize, String> = HashMap::new();
-    input.insert(0, String::from("abc"));
-    input.insert(1, String::from("def"));
-    let payload = ZBytes::from(input.clone());
-    let output = payload.deserialize::<HashMap<usize, String>>().unwrap();
-    assert_eq!(input, output);
+    // let encoding = Encoding::ZENOH_STRING;
 
     // JSON
-    let data = r#"
-    {
+    let input = serde_json::json!({
         "name": "John Doe",
         "age": 43,
         "phones": [
             "+44 1234567",
             "+44 2345678"
         ]
-    }"#;
-    let input: serde_json::Value = serde_json::from_str(data).unwrap();
-    let payload = ZBytes::try_serialize(input.clone()).unwrap();
-    let output: serde_json::Value = payload.deserialize().unwrap();
+    });
+    let payload = ZBytes::from(serde_json::to_vec(&input).unwrap());
+    let output: serde_json::Value = serde_json::from_slice(&payload.to_bytes()).unwrap();
     assert_eq!(input, output);
     // Corresponding encoding to be used in operations like `.put()`, `.reply()`, etc.
     // let encoding = Encoding::APPLICATION_JSON;
 
-    // YAML
-    let data = r#"
-        name: "John Doe"
-        age: 43
-        phones:
-          - "+44 1234567"
-          - "+44 2345678"
-    "#;
-    let input: serde_yaml::Value = serde_yaml::from_str(data).unwrap();
-    let payload = ZBytes::try_serialize(input.clone()).unwrap();
-    let output: serde_yaml::Value = payload.deserialize().unwrap();
-    assert_eq!(input, output);
-    // Corresponding encoding to be used in operations like `.put()`, `.reply()`, etc.
-    // let encoding = Encoding::APPLICATION_YAML;
-
-    // Protobuf
-    use prost::Message;
-    #[derive(Message, Eq, PartialEq)]
-    struct EntityInfo {
-        #[prost(uint32)]
-        id: u32,
-        #[prost(string)]
-        name: String,
+    // Protobuf (see example.proto)
+    mod example {
+        include!(concat!(env!("OUT_DIR"), "/example.rs"));
     }
-    let input = EntityInfo {
+    use prost::Message;
+    let input = example::Entity {
         id: 1234,
         name: String::from("John Doe"),
     };
     let payload = ZBytes::from(input.encode_to_vec());
-    let output =
-        EntityInfo::decode(Cursor::new(payload.deserialize::<Cow<[u8]>>().unwrap())).unwrap();
+    let output = example::Entity::decode(&*payload.to_bytes()).unwrap();
     assert_eq!(input, output);
     // Corresponding encoding to be used in operations like `.put()`, `.reply()`, etc.
     // let encoding = Encoding::APPLICATION_PROTOBUF;
+
```

Comment:
This example is less complete than the removed one. No examples of usage `ZSerializer` / `ZDeserializer` wrriter wrapping types.
I  suppose that writing real code with new API may cause changes in the API

Reply:
Just didn't took the time to write them.